src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -227,14 +230,9 @@ public function getProperty($name)
      */
     public function getProperties($filter = null)
     {
-        $properties = $this->betterReflectionClass->getProperties();
-
-        $wrappedProperties = [];
-        foreach ($properties as $key => $property) {
-            $wrappedProperties[$key] = new ReflectionProperty($property);
-        }
-
-        return $wrappedProperties;
+        return array_values(array_map(static function (BetterReflectionProperty $property): ReflectionProperty {
```

Comment:
Note: if you are wondering about tests, this code is already covered by tests. The change is purely type-driven, so no test additions should really be needed.